src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunTableManager.cs

```diff
@@ -53,11 +51,24 @@ protected virtual void Graph_NewMarkedNode(DependencyNodeCore<NodeFactory> obj)
 
             if (methodNode != null)
             {
-                _methodsGenerated.Add(methodNode.Method);
+                Debug.Assert(_methodsGenerated != null);
```

Comment:
This assert is no longer valid. _methodsGenerated will never be null. You should assert instead that _sortedMethods is not true

Reply:
Good catch